test/demo/utils/hydrateNotification.ts

```diff
@@ -0,0 +1,31 @@
+import {NotificationOptions, Notification} from '../../../src/client';
+
+export function assertHydratedCorrectly(options: NotificationOptions, fullNote: Notification): void {
+    // Handle `id` and `date` seperately as these cannot be predicted if undefined
+    if (options.id) {
+        expect(fullNote.id).toEqual(options.id);
+    }
+    if (options.date) {
+        expect(fullNote.date).toEqual(options.date);
+    }
+
+    // customData is never changed in any way by the provider
+    expect(fullNote.customData).toEqual(options.customData);
+
+    // Hydrate the buttons array if specified
+    if (Array.isArray(options.buttons)) {
+        options.buttons = options.buttons.map(btn => ({
+            title: btn.title,
+            iconUrl: btn.iconUrl || ''
+        }));
+    }
+
+    const expectedValues = {
+        body: options.body,
+        title: options.title,
+        subtitle: options.subtitle || '',
+        icon: options.icon || '',
+        buttons: options.buttons || []
```

Comment:
Actually, I'd be very tempted to drop the above block and write:

    options.buttons ? options.buttons.map(btn => ({...btn, iconUrl: btn.iconUrl || ''})) : []

My headcanon is that function params are `const` and immutable, so always feels weird for me when anything goes against that

Reply:
I did it this way to avoid having a long ternary like that in the middle of an object literal. I'd say that would be more confusing than what we have now.

Reply:
👍